ynput/ayon-launcher
# AYON: Bundle distribution

## Changelog Description
Since AYON server 0.3.0 are addon versions defined by bundles which affects how addons, dependency packages and installers are handled. Only source of truth, about any version of anything that should be used, is server bundle.

## Additional info
This approach required to change how we handle distribution. The only information that is defined by server is bundle name, the rest must be done manually by distribution logic. These changes are breaking compatibility with older AYON server versions.

Added UI that is used in cases a bundle that should be used is not available on server, or server does not have any bundle to use. In that case there is nothing to do and the process is killed.

## TODOs
- validate and deploy installer version - future PRs

## Testing notes:
1. Make sure you have AYON server at least 0.3.1 (0.3.0 had issues with settings).
2. Make sure openpype addon have changes from [this PR](https://github.com/ynput/OpenPype/pull/5209)
3. Run AYON launcher.
4. Addons should be loaded by the bundle defined on server as production/staging. Or by bundle name defined in `AYON_BUNDLE_NAME` environment variable.
    - if there is any bundle on server marked as production a popup should show
    - if you define `AYON_BUNDLE_NAME` env to non-existing bundle a popup should show
